docs/tutorial/appendix/source-JSON-YAML/index.md

```diff
@@ -0,0 +1,274 @@
+import uuid from "uuid"
+import Layout from "../components/layout"
+import SEO from "../components/seo"
+import { Link } from "gatsby"
+const basicTemplate = props => {
+  const { pageContext } = props
+  const { pageContent, links } = pageContext
+
+  return (
+    <Layout>
+      <SEO
+        title={`page created using basic json`}
+        keywords={[`gatsby`, `application`, `react`]}
+      />
+      <div>
+        {pageContent.map(data => (
+          <div key={uuid.v4()}>{data.item}</div>
+        ))}
+      </div>
+      <div>
+        {links.map(item => (
+          <div key={`linked_${uuid.v4()}`}>
+            <Link to={item.to}>{item.to}</Link>
+          </div>
+        ))}
+      </div>
+    </Layout>
+  )
+}
+export default basicTemplate
+```
+
+The above code is a plain React presentational component that recieves some data via the `pageContext` prop and shows it.
+
+Once all is done, issuing `gatsby develop` and opening your favourite browser of choice and navigating into:
+`http://localhost:8000/page1` for instance will what was done here.
+
+Now onto using a YAML file.
+
+### Using YAML as a source for building a site
+
+In order the same approach as the one documented above, but this time with a YAML file, is similar, in this particular case copy the YAML file located [here](https://github.com/gatsbyjs/gatsby/tree/master/examples/sourcing-from-JSON-or-Yaml/content/data/yaml/) to the `yaml` folder created earlier.
+
+Now change the `gatsby-node.js` to the following:
+
+```javascript
+const fs = require("fs")
+const yaml = require("js-yaml") // the yaml parser being imported to be used.
+exports.createPages = ({ actions }) => {
+  const { createPage } = actions
+  return new Promise(resolve => {
+    //yaml file loaded here
+    const ymlDoc = yaml.safeLoad(
+      fs.readFileSync("./content/data/yaml/index.yaml", "utf-8")
+    )
+    //
+    ymlDoc.forEach(element => {
+      createPage({
+        path: element.path,
+        component: require.resolve("./src/templates/basicTemplate.js"),
+        context: {
+          pageContent: element.content,
+          links: element.links,
+        },
+      })
+    })
+    resolve()
+  })
+}
+```
+
+The code above is somewhat similar to the code described above.
+
+The only diference here is the way how the data is loaded, here is where the `js-yaml` package installed earlier comes into play. It will load the contents of the file and create the website structure and some contents.
+
+### Using JSON as a content provider a page
+
+Now we'll go back to JSON, to be used as a source for content to a particular gatsby page.
+
+Now you'll see the reason of the `static` folder.
+Copy this [file](https://github.com/gatsbyjs/gatsby/tree/master/examples/sourcing-from-JSON-or-Yaml/static/second.json) inside the newly created folder.
+
+Create a new file inside the `pages` folder, for this example with will be named `jsononclient.js`, but whatever name you decide is fine.
```

Comment:
This could be a little smoother: "Create a new file inside the `pages` folder. For this example it will be named..."